graph/preprocessor.go

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/* Alias preprocessor:
+* The set of elements here are meant to process the alias definition portion of task.yaml
+* files. This is done by unmarshaling these elements which will then be added in a hierarchical
+* manner. Note the input must still be valid Yaml.
+*
+* Existing issues: Once the preTask is parsed out and the definitions are resolved, the read in
+* yaml will be processed to include the appropriate aliases, however this will include the previously
+* parsed in definitions which is not as efficient.
+*
+ */
+
+package graph
```

Comment:
Could you add a TODO for the default global alias mapping, i.e. `$az => azure-cli:someTag` - this is a p0 on the initial implementation

Reply:
Yes, it might also be good to meet to see which global ones we should be providing since I have less context on what images people are usually using for example